### DISCUSSION TITLE
How to obtain the writing direction for a given locale?

### DISCUSSION THREAD

**Discussion by rizzen-yazston** [unicode-org/icu4x#4521] [Q&A] [answered]
For the projects I am working on I require to know what the writing direction for a given locale in order to layout various widgets correctly according to the language of the locale. Thus wondering if there a way to obtain the writing direction (usually to left or to right in a downward direction), and if possible how would one obtain this piece of information for the locale.

PS: Can't wait for `displayname` crate to be stable. Going to be nice to display the language name in the language of the locale, instead of the language tag (locale).

**Comment by zbraniecki** [comment, marked as answer]
Hi, thanks for reaching out!

The functionality you're looking for is currently exposed in `locid_transform` crate as [`LocaleDirectionality`](https://docs.rs/icu/latest/icu/locid_transform/struct.LocaleDirectionality.html).

Notice, that this only currently supports RTL and LTR, and [does not cover TTB or BTT](https://www.unicode.org/reports/tr35/tr35-general.html#Layout_Elements).

**Reply by rizzen-yazston** [reply]
At least RTL and LTR is support in the TTB line direction, which supports most scripts.
